src/ast/semantic_analyser.cpp

```diff
@@ -277,34 +278,44 @@ void SemanticAnalyser::visit(Call &call)
     call.type = SizedType(Type::count, 8);
   }
   else if (call.func == "sum") {
+    bool sign = false;
     check_assignment(call, true, false);
-    check_nargs(call, 1);
-
-    call.type = SizedType(Type::sum, 8);
+    if (check_nargs(call, 1)) {
+      sign = call.vargs->at(0)->type.is_signed;
+    }
+    call.type = SizedType(Type::sum, 8, sign);
   }
   else if (call.func == "min") {
+    bool sign = false;
```

Comment:
Shouldn't the default be `true` (making the type signed)?

Reply:
The idea was to only make it signed when the argument is signed. That way you gain a bit when summing really large numbers. I can't think of a practical use case though so always having it signed works too.